stage #0/module #4 Algorithms, Data Structures/README.md

```diff
@@ -0,0 +1,28 @@
+<h1 align="center">Algorithms. Data structures</h1>
+
+<p align="center"><img src ="media/algorithm_logo.jpg" alt="Algorithm Logo" width = 30%></p>
+
+## Contents
+
+1. [Intro to algorithms](../module%20%234%20Algorithms%2C%20Data%20Structures/lesson%20%2301%20Intro%20to%20algorithms/README.md)
+2. [Algorithmic complexity. Big-O notation](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2302%20Algorithmic%20complexity.%20Big-O%20notation/README.md)
+3. [Linear search. Binary search](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2303%20Linear%20search.%20Binary%20search/README.md)
+4. [Data structures (Intro)](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2304%20Data%20structures%20(Intro)/README.md)
+5. [Data structures (Linear)](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2305%20Data%20structures%20(Linear)/README.md)
+6. [Data structures (Non-Linear)](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2306%20Data%20structures%20(Non-Linear)/README.md)
+7. [Recursion](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2307%20Recursion/README.md)
+8. [Sorting algorithms](../module%20%234%20Algorithms%2C%20Data%20Structures//lesson%20%2308%20Sorting%20algorithms/README.md)
+
+## Algorithms
+
+A fundamental understanding of various algorithms and data structures better equips a software developer to use them properly. Software Engineering is learnt and mastered through practice, so an examination of different problem-solving techniques better prepares you to handle challenges in the future. Eventually, you begin to recognize patterns and that helps you transfer your knowledge and understanding across various domains and problem sets.
+
+Algorithms are just like Granny's recipe book that contains all the steps required to make that special delicacy. In more technical terms, they are “well-defined **computational procedures** that takes some value, or set of values, as **input** and produces some value, or set of values as **output**.
+
+One thing that makes algorithms worth studying is that in most cases, there exists more than one algorithm capable of handling the task at hand as in the library case above. Again in this context, efficiency is key and is determined by considering Time complexity(the execution time of an operation) and Space complexity(memory usage of an operation) of an algorithm.
```

Comment:
I believe that it would be nice to rewrite materials, which are copied as a raw text from the Internet: https://www.better.dev/a-gentle-introduction-to-algorithms-and-data-structures.